src/gui/qgsadvanceddigitizingdockwidget.h

```diff
@@ -368,6 +370,77 @@ class GUI_EXPORT QgsAdvancedDigitizingDockWidget : public QgsDockWidget, private
      */
     void pointChanged( const QgsPointXY &point );
 
+    //! Signals for external widgets that need to update according to current values
+
+    /**
+    * Emitted whenever the value changes (either the mouse moved, or the user changed the input).
+    * Could be used by widgets that must reflect the current advanced digitizing state.
+    *
+    * \param value The current value (as a QString, as it could be an expression).
+    */
+    void valueXChanged( QString value );
+    void valueYChanged( QString value );
+    void valueAngleChanged( QString value );
+    void valueDistanceChanged( QString value );
+
+    /**
+    * Emitted whenever the parameter is locked.
+    * Could be used by widgets that must reflect the current advanced digitizing state.
+    *
+    * \param locked Whether the parameter is locked or not.
+    */
```

Comment:
Doc strings always apply to a single method, so this needs to be copied to every of the 4 methods of the block below. (It's for the generated C++ and Python API docs).

Reply:
Ok. It seems doxygen supports one docstring for multiple fcuntions (https://stackoverflow.com/a/37286334) but it doesn't seem used in QGIS... Would make the code more readable IMO.

Reply:
Looks nice.
But it would need some (non trivial) update to sipify (the sip code generator) too. I'm not sure you want to start going down that hole ;)

Reply:
Ok, copies the docstring to all methods